middleware/CompressionHandler.md

```diff
@@ -1,47 +1,64 @@
 # Compression Handler
 
-A middleware component that requests, detects and decompresses response bodies.
+A middleware component that compresses request bodies and falls back to an uncompressed body if the API doesn't support request body compression.
 
 ## Requirements
 
-- Add `Accept-encoding: gzip` to request headers.
-- Decompress response bodies that have the header `Content-Encoding: gzip`
+- Add `Content-encoding: gzip` to request headers for requests with a body.
+- Compresses request bodies.
+- Should the handler receive a [415 Unsupported Media Type](https://httpwg.org/specs/rfc7231.html#status.415) response or a `400 Bad Request` response with an OData error containing the following message `Unable to read JSON request payload. Please ensure Content-Type header is set and payload is of valid JSON format.` when the content-type was `application/json`, the handler will retry sending the request without compressing it a maximum of one time before passing on the error.
```

Comment:
Retry is not going to be possible if the request body is based on an unbuffered stream that can't be rewound.  We should fail in a similar way that we do in the retry handler in this case.

Reply:
so no retry at all? or should we try to detect whether the stream is buffered?

Reply:
additionally, after more discussions with the AGS folks, it looks like the change is also going to roll out to national clouds shortly after. So I'm wondering whether we should delay our rollout until after theirs is done so we can have that for error handling instead

- Should the handler receive a [415 Unsupported Media Type](https://httpwg.org/specs/rfc7231.html#status.415) response and the request body stream is replayable/buffered, the handler will retry sending the request without compressing it a maximum of one time before passing on the error.

Thoughts?